### ISSUE TITLE
Error loading model

### ISSUE THREAD

**Issue by zarmonela** [Macoron/gpt4all.unity#14]
hey getting this error please help 

NullReferenceException: Object reference not set to an instance of an object
Gpt4All.LlmManager.InitModel () (at ./Packages/com.gpt4all.unity/Scripts/LlmManager.cs:114)
UnityEngine.Debug:LogException(Exception)
Gpt4All.<InitModel>d__37:MoveNext() (at ./Packages/com.gpt4all.unity/Scripts/LlmManager.cs:121)
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:SetResult(LlmWrapper)
Gpt4All.<InitFromPathAsync>d__28:MoveNext() (at ./Packages/com.gpt4all.unity/Scripts/LlmWrapper.cs:214)
UnityEngine.UnitySynchronizationContext:ExecuteTasks()

**Comment by zarmonela** [comment]
LLM model path C:/Users/Bz/Downloads/gpt4all.unity-master (1)/gpt4all.unity-master/Assets/StreamingAssets\Assets/StreamingAssets/Gpt4All/wizardlm-13b-v1.1-superhot-8k.ggmlv3.q4_0.bin doesn't exist!
UnityEngine.Debug:LogError (object)
Gpt4All.LlmWrapper:InitFromPath (Gpt4All.LlmModelType,string) (at ./Packages/com.gpt4all.unity/Scripts/LlmWrapper.cs:166)
Gpt4All.LlmWrapper/<>c__DisplayClass28_0:<InitFromPathAsync>b__0 () (at ./Packages/com.gpt4all.unity/Scripts/LlmWrapper.cs:212)
System.Threading._ThreadPoolWaitCallback:PerformWaitCallback ()

**Comment by danger-electrodes** [comment]
Hello, I'm getting the same error as OP as well as an error due to Whisper? 

Model not loaded error : 

NullReferenceException: Object reference not set to an instance of an object
Gpt4All.LlmManager.InitModel () (at Packages/com.gpt4all.unity/Scripts/LlmManager.cs:114)
UnityEngine.Debug:LogException(Exception)
Gpt4All.<InitModel>d__37:MoveNext() (at Packages/com.gpt4all.unity/Scripts/LlmManager.cs:121)
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:SetResult(LlmWrapper)
Gpt4All.<InitFromPathAsync>d__28:MoveNext() (at Packages/com.gpt4all.unity/Scripts/LlmWrapper.cs:214)
UnityEngine.UnitySynchronizationContext:ExecuteTasks()

Whisper Error :

Whisper model isn't loaded! Init Whisper model first!
UnityEngine.Debug:LogError (object)
Gpt4All.LlmManager/<CheckIfLoaded>d__39:MoveNext () (at Packages/com.gpt4all.unity/Scripts/LlmManager.cs:142)
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<bool>:Start<Gpt4All.LlmManager/<CheckIfLoaded>d__39> (Gpt4All.LlmManager/<CheckIfLoaded>d__39&)
Gpt4All.LlmManager:CheckIfLoaded ()
Gpt4All.LlmManager/<Prompt>d__38:MoveNext () (at Packages/com.gpt4all.unity/Scripts/LlmManager.cs:129)
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<string>:Start<Gpt4All.LlmManager/<Prompt>d__38> (Gpt4All.LlmManager/<Prompt>d__38&)
Gpt4All.LlmManager:Prompt (string)
Gpt4All.Samples.ChatSample/<SendToChat>d__9:MoveNext () (at Assets/Samples/1 - Chat/ChatSample.cs:47)
System.Runtime.CompilerServices.AsyncVoidMethodBuilder:Start<Gpt4All.Samples.ChatSample/<SendToChat>d__9> (Gpt4All.Samples.ChatSample/<SendToChat>d__9&)
Gpt4All.Samples.ChatSample:SendToChat (string)
Gpt4All.Samples.ChatSample:OnSubmitPressed () (at Assets/Samples/1 - Chat/ChatSample.cs:35)
UnityEngine.EventSystems.EventSystem:Update () (at Library/PackageCache/com.unity.ugui@1.0.0/Runtime/EventSystem/EventSystem.cs:501)

Do I need to add your whisper package too?

Could you also provide a link to the models you mentionned too? I can't find them on gpt4all.io?

**Comment by Macoron** [comment]
> Hello, I'm getting the same error as OP as well as an error due to Whisper?
> Do I need to add your whisper package too?

No, it's just a typo. You don't need whisper to run LLMs. Actual error is just saying that it failed to load model weights for some reason.

> Could you also provide a link to the models you mentionned too? I can't find them on gpt4all.io?

gpt4all deprecated old models recently. I make fork of their models on HF. [Check readme](https://github.com/Macoron/gpt4all.unity/blob/9045baeed463409a5e197aa58dc33ff1c26a0c5f/README.md) to get compatible models weights.

**Comment by zarmonela** [comment]
> Wizardlm isn't supported by current version of gpt4all-unity. Right now it was tested with:
> 
> * mpt-7b-chat
> * gpt4all-j-v1.3-groovy.bin
> * gpt4all-l13b-snoozy
> 
> Check #11 for more information

Hey! I'm working on updating the project to incorporate the new bindings. I'd love to chat and ask you a few questions if you're available. Please let me know.

**Comment by Macoron** [comment]
@zarmonela sure, I created a quick Discord group, so we can discuss it there.
https://discord.gg/RAjD5yY7bP

Alternatively, you can ask all questions in this issue.

**Comment by Elevons** [comment]
How's this going? Any way I can help?
